src/Lecture2.hs

```diff
@@ -74,7 +86,16 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt n (x:xs)
+  | n < 0 = (Nothing , x:xs)
```

Comment:
You can implement the main body in a helper recursive function to avoid the `< 0` on every recursive call. You need to check if `n < 0` only once in the beginning 🙂